lib/ddtrace/utils.rb

```diff
@@ -1,9 +1,21 @@
+require 'thread'
+
 module Datadog
-  # TODO[manu]: write docs
+  # Utils contains low-level utilities, typically to provide pseudo-random trace IDs.
   module Utils
+    # We use a custom random number generator because we want no interference
+    # with the default one. Using the default prng, we could break code that
+    # would rely on srand/rand sequences.
+    @rnd = Random.new
+    # We want to garantee all calls are sequential else, in some edge cases,
+    # the same value could be return twice, ending in trace collision.
```

Comment:
in real multi-threading languages of course you need a lock for a shared memory access, but the GIL ensures no collisions. About removing the GIL in Ruby 3 we'll see, since it's definitely something really hard to remove and that breaks any Ruby application.